### ISSUE TITLE
[Feature] Go To Definition does not work or is not fully resolved

### ISSUE THREAD

**Issue by DrakeEvans** [NomicFoundation/hardhat-vscode#553]
Feature: Go To Definition should support function definitions as well as fully resolve Definitions

Current Situation: Function definitions not supported, For supported symbols Go To Definition only goes to the nearest declaration not to the implementation

https://github.com/NomicFoundation/hardhat-vscode/assets/31104161/67496b24-9d93-4ea3-97ae-ecf30fd41b18

Proposed: Function definitions supported, Go To Definition fully resolves to implementation not nearest declaration

https://github.com/NomicFoundation/hardhat-vscode/assets/31104161/bf6ba5e8-30c1-4ae2-896d-a1fbfd3a0e72

**Comment by DrakeEvans** [comment]
@kanej Yes you can see it at https://github.com/agora-finance/agora-dollar-evm-tests/blob/master/src/test/erc20-privilaged/TestMinting.t.sol

You will need to run forge install and npm i to make it work

**Comment by DrakeEvans** [comment]
Will need to run npm i inside of one submodule as well `cd lib/agora-dollar-evm/ && npm i``

**Comment by DrakeEvans** [comment]
> so there is no way to fix this thing? I experiencing the same with almost every project

Im not a contributor but this may be fixed as they move to a different syntax tree parser.  You can check out the slaang project

**Comment by kanej** [comment]
> > so there is no way to fix this thing? I experiencing the same with almost every project
> 
> Im not a contributor but this may be fixed as they move to a different syntax tree parser. You can check out the slaang project

That is correct. We are rolling out Slang internally within the extension (it is currently used to support the outline view).

The Slang project is working on Solidity lang binding information to enable go to definition type use cases. Once we have integrated Slang's binding info and its more robust project analysis, there should be significant improvements in our navigation support.
